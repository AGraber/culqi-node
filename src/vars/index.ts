export default {
  baseEndpoint: {
    domain: 'api.culqi.com',
    basePath: '/v2',
  },
  baseSecureEndpoint: {
    domain: 'secure.culqi.com',
    basePath: '/v2',
  },
  basePaths: {
    tokens: '/tokens',
    charges: '/charges',
    refunds: '/refunds',
    plans: '/plans',
    subscriptions: '/subscriptions',
    customers: '/customers',
    cards: '/cards',
    events: '/events',
  },
  publicKey: '',
  privateKey: '',
};
