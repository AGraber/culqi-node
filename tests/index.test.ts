import Culqi from '../src/index';
import vars from '../src/vars';

describe('Culqi', () => {
  it('should throw error if privateKey is not set', () => {
    try {
      // tslint:disable-next-line:no-unused-expression
      new Culqi();
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });

  it('should set private key', () => {
    // tslint:disable-next-line:no-unused-expression
    new Culqi({
      privateKey: 'private-key-test',
    });
    expect(vars.privateKey).toBe('private-key-test');
  });

  it('should throw error if pciCompliant is true but no public key was provided', () => {
    try {
      // tslint:disable-next-line:no-unused-expression
      new Culqi({
        privateKey: 'abc',
        pciCompliant: true,
      });
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });

  it('should throw error if publicKey is provided but pciCompliant is not true', () => {
    try {
      // tslint:disable-next-line:no-unused-expression
      new Culqi({
        privateKey: 'abc',
        publicKey: 'abc',
      });
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });

  it('should set privateKey and publicKey', () => {
    // tslint:disable-next-line:no-unused-expression
    new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(vars.privateKey).toBe('abc');
    expect(vars.publicKey).toBe('bcd');
  });

  it('should have tokens property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.tokens).toMatchSnapshot();
  });

  it('should have charges property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.charges).toMatchSnapshot();
  });

  it('should have refunds property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.refunds).toMatchSnapshot();
  });
});
