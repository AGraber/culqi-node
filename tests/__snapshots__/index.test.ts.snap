// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Culqi should have charges property 1`] = `
Object {
  "captureCharge": [Function],
  "createCharge": [Function],
  "getCharge": [Function],
  "getCharges": [Function],
  "updateCharge": [Function],
}
`;

exports[`Culqi should have refunds property 1`] = `
Object {
  "createRefund": [Function],
  "getRefund": [Function],
  "getRefunds": [Function],
  "updateRefunds": [Function],
}
`;

exports[`Culqi should have tokens property 1`] = `
Object {
  "createToken": [Function],
  "getToken": [Function],
  "getTokens": [Function],
  "updateToken": [Function],
}
`;

exports[`Culqi should throw error if pciCompliant is true but no public key was provided 1`] = `[Error: Provide 'publicKey' property]`;

exports[`Culqi should throw error if privateKey is not set 1`] = `[Error: Provide 'privateKey' property]`;

exports[`Culqi should throw error if publicKey is provided but pciCompliant is not true 1`] = `[Error: Provide 'pciCompliant' options as true. It means you acknowledge that you are PCI-compliant in order to execute certain operations.]`;
